src/Symfony/Component/DependencyInjection/Compiler/ResolveInvalidReferencesPass.php

```diff
@@ -71,16 +71,17 @@ public function process(ContainerBuilder $container)
      *
      * @param array $arguments    An array of Reference objects
      * @param bool  $inMethodCall
+     * @param bool  $inCollection
      *
      * @return array
      *
      * @throws RuntimeException When the config is invalid
      */
-    private function processArguments(array $arguments, $inMethodCall = false)
+    private function processArguments(array $arguments, $inMethodCall = false, $inCollection = false)
     {
         foreach ($arguments as $k => $argument) {
```

Comment:
I suggests doing `$isNumeric = array_keys($arguments) !== range(0, count($arguments) - 1);` before the foreach